server/controllers/Messages/resolver.ts

```diff
@@ -0,0 +1,19 @@
+import { Resolver, Mutation, Arg } from 'type-graphql';
+import MailerService from '../../services/MailerService';
+import { SendEmailInputs } from './inputs';
+import { Email } from './Email';
+
+@Resolver()
+export class EmailResolver {
+  @Mutation(() => Email) async sendEmail(@Arg('data') data: SendEmailInputs) {
+    const email = new MailerService(data.to, data.subject, data.html);
+    await email.sendEmail();
+    return {
+      ourEmail: email.ourEmail,
+      emailList: email.emailList,
+      subject: email.subject,
+      htmlEmail: email.htmlEmail,
+      backupText: email.backupText,
+    };
+  }
+}
```

Comment:
```suggestion
@Resolver()
export class EmailResolver {
  @Mutation(() => Email) async sendEmail(@Arg('data') data: SendEmailInputs) {
    const email = new MailerService(data.to, data.subject, data.html);
    const errors = await validate(data);

    if (errors.length > 0) {
      console.log('validation failed. errors: ', errors);
      // This object is returned just to pass the type checks.
      return {
        ourEmail: 'this failed', // Debugging entry, so you can see from the
        // graphql playground that the validation failed
        emailList: email.emailList,
        subject: email.subject,
        htmlEmail: email.htmlEmail,
        backupText: email.backupText,
      };
    } else {
      console.log('validation succeeded');
      await email.sendEmail();
      return {
        ourEmail: email.ourEmail,
        emailList: email.emailList,
        subject: email.subject,
        htmlEmail: email.htmlEmail,
        backupText: email.backupText,
      };
    }
  }
}
```
This (obviously) isn't a real solution, but I hope it helps.

Reply:
@ojeytonwilliams It appears that the input validation wasn't working because the 'validate: false' option was set in the buildSchema() function in server/app.ts. Once I remove the 'validate: false' option the built in validation works. I prefer the built in method over importing 'validate' from 'class-validator' and writing custom error handling logic in the resolver, but I suppose the option was set for a reason. @Zeko369 Is there a specific reason the 'validate: false' option is set in the buidSchema() function?

Reply:
Ah, I assumed we needed to invoke `validate` directly.  I'm glad you spotted that.

From what I can gather from the [docs](https://typegraphql.com/docs/validation.html#how-to-use) you only use that option if you want to manually enable validation for specific arguments.  Given that `SendEmailInputs` is the only class being validated at the moment, it should be fine.

Reply:
According to the docs input validation is enabled by default, all that is needed is to import class-validator and decorate the fields with the appropriate validation requirements. The 'validate: false' option is used to disable the default validation. At the moment the SendEmailInputs class is the only class validating input, but whichever method we use to validate input should be used in any resolver that accepts inputs, so we should decide early on which method we're going to use so we can be consistent. I prefer using the default validation, unless someone else disagrees.

Reply:
It seems to me that custom handling approach would be much slower to write, so we might as well defer that effort until we're confident it's necessary.  I'd say go ahead with the default validation for now and we can always switch to the custom approach later if we have to.